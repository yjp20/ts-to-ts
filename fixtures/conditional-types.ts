/** @model */
type IsString<T> = T extends string ? true : false;

/** @model */
type Flatten<T> = T extends Array<infer U> ? U : T;

/** @model */
type NonNullable<T> = T extends null | undefined ? never : T;

/** @model */
type ExtractStrings<T> = T extends string ? T : never;
