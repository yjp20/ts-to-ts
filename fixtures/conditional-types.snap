<<<<<<<<<<< typescript <<<<<<<<<<<<
/** @model */
type IsString<T> = T extends string ? true : false;

/** @model */
type Flatten<T> = T extends Array<infer U> ? U : T;

/** @model */
type NonNullable<T> = T extends null | undefined ? never : T;

/** @model */
type ExtractStrings<T> = T extends string ? T : never;
===================================
// conditional-types.ts
alias IsString = IsString<T>;

alias Flatten = Flatten<T>;

alias NonNullable = NonNullable<T>;

alias ExtractStrings = ExtractStrings<T>;
>>>>>>>>>>> typespec >>>>>>>>>>>>>>